### PR TITLE
Add open source OIDC providers to security page

### LINKED ISSUE #2687: Document open source alternatives to closed-source OIDC providers
I think it's great that Origin Authentication works with JSON Web Tokens and that it is documented that closed source providers like Auth0, Google Auth, and Firebase Auth are compatible with this. I want to suggest to include well-adopted open source alternatives such as [Keycloak](https://www.keycloak.org) and [ORY Hydra](https://github.com/ory/hydra) as well. Both are OpenID Connect Certified for all [relevant OIDC flows](https://openid.net/certification/).

If this is to your liking, I'm happy to supply a PR.

### PR DESCRIPTION
**Pull request by aeneasr**
Adds open source OIDC providers ORY Hydra and Keycloak to the security overview page.

Closes #2687

### CODE REVIEW

**Comment by aeneasr** [comment]
I signed the CLA some time ago but have since changed my username from aeneasr. Anyways, I updated the information in the CLA form so it should now be good to go!

File: content/docs/concepts/security/index.md [hunk 1]
```diff
@@ -224,7 +224,7 @@ Istio provides two types of authentication:
 - **Origin authentication**, also known as **end-user authentication**: verifies the
   original client making the request as an end-user or device.
   Istio enables request-level authentication with JSON Web Token (JWT) validation
-  and a streamlined developer experience for [Auth0](https://auth0.com/), [Firebase Auth](https://firebase.google.com/docs/auth/),
+  and a streamlined developer experience for open source OpenID Connect providers [ORY Hydra](https://www.ory.sh), [Keycloak](https://www.keycloak.org), closed source providers [Auth0](https://auth0.com/), [Firebase Auth](https://firebase.google.com/docs/auth/),
```
Comment (aeneasr) [inline content/docs/concepts/security/index.md]: I'm not sure if we should explicitly call out OpenID Connect here as we're really allowing **any** JWT. Also I'm not sure if OIDC is the right choice here as we're typically using it on the client side (relying party / RP) as opposed to the resource provider (RP). There we would probably use opaque or transparent bearer tokens such as OAuth 2.0 Access Tokens formatted as random strings (opaque) or as JWTs (transparent).

In fact, I think this is something which is lacking in the documentation of itsio. I'm still getting into the whole topic but am a bit confused about end-user authentication and authorization. This is probably not the right issue to discuss this but I'd be happy to open another one for this.

**Comment by aeneasr** [comment]
@geeknoid anything I can do to get this merged? Failing tests seem to be flaky and not related to this PR.

**Comment by geeknoid** [comment]
These two errors are from your changes:

    content/docs/concepts/security/index.md
      227 | er experience for open source OpenID Connect providers [ORY Hydra] 
      227 |  Hydra](https://www.ory.sh), [Keycloak](https://www.keycloak.org), c 

See the README.md file at the root of the repo to learn what to do with spelling errors.

File: content/docs/concepts/security/index.md (hunk 1 above)
Comment (liminw) [reply content/docs/concepts/security/index.md]: It's a good point that currently Istio only provides JWT authentication. OIDC connect is more on the client side to acquire a token. There is an ongoing work in Istio community to provide OIDC (token acquisition) support at Ingress. For now, it's probably more accurate to say that
"Istio enables request-level authentication with JSON Web Token (JWT) validation for JWTs issued by any OpenID Connect providers, such as [Auth0](https://auth0.com/), [Firebase Auth](https://firebase.google.com/docs/auth/), [Google Auth](https://developers.google.com/identity/protocols/OpenIDConnect), [ORY Hydra](https://www.ory.sh), [Keycloak](https://www.keycloak.org). "

I am curious about why you think "JWT authorization is not sufficient for most end-user sessions". Istio authorization allows customers to authorize any JWT claims today. We could chat more on this (my email is liminwang@google.com). Or you can open a GitHub issue for that.

**Comment by aeneasr** [comment]
Ok, spellcheck is fixed, test is still failing but this is not the patch's fault (this time for real) :)

**Pull request merged**